### ISSUE TITLE
Filter on elements (e.g. :enabled, :disabled, :hidden, :visible)

### ISSUE THREAD

**Issue by austek-mdsol** [webdriverio/webdriverio#829]
Is it possible to filter on what is returned.

Currently, I am trying to select some elements and run tests only on enabled ones

```
browser.elements('.body_container .btn.btn-primary', function(err, results) {
      results.value.forEach(function(elem) {
        browser.elementIdAttribute(elem.ELEMENT,'disabled').then(function(disabled){
          if(!disabled.value) {
            browser.elementIdCssProperty(elem.ELEMENT, 'background-color').then(function(bgColor) {
                try {
                  failed = false;
                  console.log('bgColor: ' + JSON.stringify(bgColor));
                  assert(bgColor.parsed.hex === '#0055b7');
                } catch (e) {
                  failed = true;
                }
              });
          }
        });
      });
      assert.equal(failed, false);
    })
      .call(next);
```
In above code there are two problems:
1.   `elementIdCssProperty` fails to find the element or it doesn't execute. (not sure, it doesn't print bgColor)
2. When I put `assert.equal(false, true)` above `if(!disabled.value) {` execution seems to stop but the test is not failed.

**Comment by christian-bromann** [comment]
What version of WebdriverIO are you using?

**Comment by austek-mdsol** [comment]
Version: 3.2.5

**Comment by christian-bromann** [comment]
That's a nice feature request. I guess it should be able to add this kind of functionality to our query selector implementation.

**Comment by drets** [comment]
Hey!

About `:enabled`, `:disabled`.

It's already works:

```html
	<div id="orange">
	  <button class="apple" disabled="disabled">Disabled</button>
	  <button class="apple">Enabled</button>
	</div>
```

```js
    it('should find disabled element', function() {
       return this.client.elements("#orange .apple:disabled", function(err, elements) {
            assert(!err);
            assert.equal(elements.value.length, 1);
        });
    });
```

Selenium tests for this functionality: https://github.com/SeleniumHQ/selenium/blob/ab1e647d0fc8fc39e6b00ae94321ab228b6728f2/java/client/test/com/thoughtworks/selenium/corebased/TestCssLocators.java#L130-L132

**Comment by christian-bromann** [comment]
merged with #891
